docs/permission/writing-a-policy.md

```diff
@@ -3,3 +3,137 @@ id: writing-a-policy
+
+## Conditional decisions
+
+Let's change the policy to the following:
+
+```diff
+- import { IdentityClient } from '@backstage/plugin-auth-node';
++ import { BackstageIdentityResponse, IdentityClient } from '@backstage/plugin-auth-node';
++ import { catalogConditions, createCatalogPolicyDecision } from '@backstage/plugin-catalog-backend';
+
+  ...
+
+  class TestPermissionPolicy implements PermissionPolicy {
+-   async handle(request: PolicyAuthorizeQuery): Promise<PolicyDecision> {
++   async handle(
++     request: PolicyAuthorizeQuery,
++     user?: BackstageIdentityResponse,
++    ): Promise<PolicyDecision> {
+      if (request.permission.name === 'catalog.entity.delete') {
+-       return {
+-         result: AuthorizeResult.DENY,
+-       };
++       return createCatalogPolicyDecision(
++         catalogConditions.isEntityOwner(
++           user?.identity.ownershipEntityRefs ?? [],
++         ),
++       );
+      }
+
+      return { result: AuthorizeResult.ALLOW };
+    }
+  }
+```
+
+Let's walk through the new code that we just added. Inside of the if statement, instead of returning a Definitive Policy Decision of DENY, we now return a [Conditional Policy Decision](https://backstage.io/docs/reference/plugin-permission-node.conditionalpolicydecision) (See the [Concepts page](./concepts.md) for more details). This is a way for policies to defer the evaulation of the decision back to the plugin which owns the permission. This allows the framework to support cases in which the policy does not have all the information required to make a decision.
+
+In the policy above, there's no way for the handle method to determine whether the user who is trying to unregister the entity is the owner of that entity. So we use the `createCatalogPolicyDecision` helper provided by the catalog backend to craft a conditional decision, which allows us to tell the catalog backend that it should only return ALLOW if the user owns the entity.
+
+The `catalogConditions` object contains various conditions that the catalog plug authors have provided for us to use in authoring our policy. Thankfully, they have provided the `isEntityOwner` rule, which is exactly what we need.
+
+The `isEntityOwner` rule needs a list of claims as a parameter, and the second argument to our `handle` method provides us with a `BackstageIdentityResponse` object, from which we can grab the user's `ownershipEntityRefs`. We provide an empty array as a fallback since the user may be anonymous.
+
+You should now be able to see in your Backstage app that the unregister entity button is enabled for entities that you own, but disabled for all other entities!
+
+## Resource types
+
+Now let's say we would also like to restrict users from viewing catalog entities that they do not own, just like we did for unregistering entities. One way to achieve this may be to simply duplicate our if statement and check for the `catalog.entti.read` permission:
```

Comment:
```suggestion
Now let's say we would also like to restrict users from viewing catalog entities that they do not own, just like we did for unregistering entities. One way to achieve this may be to simply duplicate our if statement and check for the `catalog.entity.read` permission:
```